### ISSUE TITLE
log_client: allow more detailed rsyslog rules configuration

### ISSUE THREAD

**Issue by DilasserT** [bluebanquise/bluebanquise#488]
Hello, 

It would be nice to have more flexibility with the rsyslog configuration. For example, I would like to change the forwarding rule: 

`
*.* @@{{networks[j2_node_main_network]['services_ip']['log_ip']}}:{{ log_client_server_port }}
`
to 
`
*.notice;kern.debug @@{{networks[j2_node_main_network]['services_ip']['log_ip']}}:{{ log_client_server_port }}
`

It can also be applied to the RULES section. 
There are a lot of possibilities, so it makes sense to me to allow the user to override the default template.
Also, if someone wants to tinker with one part while keeping the other part, it might be interesting to break down the configuration in several files ?

Something like that: 

/etc
├── rsyslog.conf
└── rsyslog.d
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;├── local-rules.conf
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;└── forwarding-rules.conf

What do you think ?

**Comment by btravouillon** [comment]
I do agree.
Any user configuration must be kept in the user's inventory, so we need to find a way to override default templates with data from the inventory. This was done recently for [the new nhc role](https://github.com/bluebanquise/bluebanquise/pull/448#discussion_r520661613).

**Comment by DilasserT** [comment]
About the default templates, would it be interesting to have that division ?
/etc
├── rsyslog.conf -> global rules
└── rsyslog.d
        ├── local-rules.conf -> node level rules
        └── forwarding-rules.conf -> forwarding rules

I am asking this because in my case, I would like to overwrite the forwarding rule while keeping the other configuration.

**Comment by btravouillon** [comment]
I think so. How would you implement it in an ansible task?

**Comment by DilasserT** [comment]
If I understand it correctly, in the nhc role there is an empty array for additional configuration files, a similar task can be implemented in log_client but with an array containing those files.

**Comment by btravouillon** [comment]
Well, the issue with the implementation in the nhc role is that the custom files must be installed in the role directory, which is not in the scope of the user inventory: this will split the configuration reference between inventory and roles.

We could implement the code below:

```
- name: template █ Copy log custom files
  template:
    src: rsyslog.d_custom.conf.j2
    dest: "/etc/rsyslog.d/{{ item }}.conf"
    owner: root
    group: root
    mode: 0644
  loop: "{{ log_client_custom_conf }}"
```

```
# cat defaults/main.yml
---
log_client_custom_conf: []
```

```
# cat templates/rsyslog.d_custom.conf.j2
#### BlueBanquise file ####
## {{ansible_managed}}

# Configuration {{ item }}

{{ hostvars[inventory_hostname]['log_client_' + item + '_conf'] }}
```

Usage in inventory (example for the group mg_managements):
```
# cat inventory/group_vars/mg_managements/log_client.yml
log_client_custom_conf:
  - local_rules
  - forwarding_rules

log_client_local_rules_conf: |
  my configuration for local-rules
  can be multiline

log_client_forwarding_rules_conf: |

  my configuration for forwarding-rules
```

Which would produce:
```
--- before
+++ after: /root/.ansible/tmp/ansible-local-238490ogxrjho/tmpjyyrgs5_/rsyslog.d_custom.conf.j2
@@ -0,0 +1,7 @@
+#### BlueBanquise file ####
+## Ansible managed file, do not edit
+
+# Configuration local_rules
+
+my configuration for local-rules
+can be multiline

changed: [management1] => (item=local_rules)
--- before
+++ after: /root/.ansible/tmp/ansible-local-238490ogxrjho/tmpddidb53c/rsyslog.d_custom.conf.j2
@@ -0,0 +1,7 @@
+#### BlueBanquise file ####
+## Ansible managed file, do not edit
+
+# Configuration forwarding_rules
+
+
+my configuration for forwarding-rules

changed: [management1] => (item=forwarding_rules)
```

Drawback: I had to replace dash with underscore since dash is not valid in variable name. It should not be an issue anyway.

**Comment by DilasserT** [comment]
It looks good to me! I can't seem to find another solution as clear and appropriate.

**Comment by oxedions** [comment]
Hello.

I also like this solution. I will adapt the NHC role to this.

**Comment by DilasserT** [comment]
@btravouillon Are you planning to do a PR for this ? I can help if you do not have time.

**Comment by btravouillon** [comment]
@DilasserT Go ahead and open a PR if possible. Thanks.

**Comment by btravouillon** [comment]
I wonder if a better approach could be a single variable:

```
log_client_configuration_files:
  - name: local-rules.conf
    content: |
      my configuration for local-rules
      can be multiline
  - name: forwarding-rules.conf
    content: |

      my configuration for forwarding-rules
```

The task:
```
- name: template █ Install custom configuration files
  template:
    src: rsyslog.d_custom.conf.j2
    dest: "/etc/rsyslog.d/{{ item.name }}"
    owner: root
    group: root
    mode: 0644
  loop: "{{ log_client_configuration_files }}"
  loop_control:
    label: "Install configuration file {{ item.name }}"
```

The template:
```
#### BlueBanquise file ####
## {{ansible_managed}}

# Configuration {{ item.name }}

{{ item.content }}
```

It would remove the burden of file name that must respect Ansible variable naming rule (no dot, no dash, ...).

**Comment by oxedions** [comment]
Jeez ! I like this one !
Clean, and everything is concentrated in a single place. 👍 I would go for that.

**Comment by btravouillon** [comment]
> I also like this solution. I will adapt the NHC role to this.

See #497 for NHC.

**Comment by btravouillon** [comment]
Fixed in #494
